libs/vre/shared/app-config/src/lib/app-config/app-config.service.ts

```diff
@@ -52,85 +50,47 @@ export class AppConfigService {
-
-        const zioPrefix =
-            this._appConfig.zioPrefix === '/zio' ||
-            this._appConfig.zioPrefix === ':5555'
-                ? this._appConfig.zioPrefix
-                : '/zio';
-        const zioEndpoints = Array.isArray(this._appConfig.zioEndpoints)
-            ? this._appConfig.zioEndpoints
-            : [];
-
         this._dspApiConfig = new KnoraApiConfig(
-            this._appConfig.apiProtocol,
-            this._appConfig.apiHost,
-            apiPort,
-            apiPath,
-            jsonWebToken,
-            logErrors,
-            zioPrefix,
-            zioEndpoints
+            c.apiProtocol,
+            c.apiHost,
+            c.apiPort,
+            c.apiPath,
+            c.jsonWebToken,
+            c.logErrors,
+            c.zioPrefix,
+            c.zioEndpoints
         );
 
-        /**
-         * make input type safe, as AppConfig can contain basically anything as
-         * config.json is simply pressed into this type without type checking!!!
-         */
-        const iiifPort =
-          typeof this._appConfig.iiifPort === 'number'
-            ? this._appConfig.iiifPort
-            : null;
-
-        const iiifPath = this._appConfig.iiifPath;
-
         // init iiif configuration
         this._dspIiifConfig = new DspIiifConfig(
-            this._appConfig.iiifProtocol,
-            this._appConfig.iiifHost,
-            iiifPort,
-            iiifPath
+            c.iiifProtocol,
+            c.iiifHost,
+            c.iiifPort,
+            c.iiifPath
         );
 
         // init dsp app extended configuration
-        this._dspAppConfig = new DspAppConfig(
-            this._appConfig.geonameToken,
-            this._appConfig.iriBase
-        );
+        this._dspAppConfig = new DspAppConfig(c.geonameToken, c.iriBase);
 
         // init instrumentation configuration
         this._dspInstrumentationConfig = new DspInstrumentationConfig(
-            this._appConfig.instrumentation.environment,
+            c.instrumentation.environment,
             new DspDataDogConfig(
-                this._appConfig.instrumentation.dataDog.enabled,
-                this._appConfig.instrumentation.dataDog.applicationId,
-                this._appConfig.instrumentation.dataDog.clientToken,
-                this._appConfig.instrumentation.dataDog.site,
-                this._appConfig.instrumentation.dataDog.service
+                c.instrumentation.dataDog.enabled,
+                c.instrumentation.dataDog.applicationId,
+                c.instrumentation.dataDog.clientToken,
+                c.instrumentation.dataDog.site,
+                c.instrumentation.dataDog.service
             ),
             new DspRollbarConfig(
-                this._appConfig.instrumentation.rollbar.enabled,
-                this._appConfig.instrumentation.rollbar.accessToken
+                c.instrumentation.rollbar.enabled,
+                c.instrumentation.rollbar.accessToken
             )
```

Comment:
This is a code smell: https://refactoring.guru/smells/long-parameter-list

Reply:
thanks, you are right. I would prefer to not include this into this PR, as it is out of scope.